internal/channelz/funcs.go

```diff
@@ -184,53 +185,76 @@ func GetServer(id int64) *ServerMetric {
 	return db.get().GetServer(id)
 }
 
-// RegisterChannel registers the given channel c in channelz database with ref
-// as its reference name, and add it to the child list of its parent (identified
-// by pid). pid = 0 means no parent. It returns the unique channelz tracking id
-// assigned to this channel.
-func RegisterChannel(c Channel, pid int64, ref string) int64 {
+// RegisterChannel registers the given channel c in the channelz database with
+// ref as its reference name, and adds it to the child list of its parent
+// (identified by pid). pid == nil means no parent.
+//
+// Returns a unique channelz identifier assigned to this channel.
+//
+// If channelz is not turned ON, this function is a no-op.
+func RegisterChannel(c Channel, pid *Identifier, ref string) *Identifier {
+	if !IsOn() {
+		return nil
+	}
+
 	id := idGen.genID()
+	parent := int64(0)
```

Comment:
`var parent int64`?
Not sure if it saves the cast, or is the same as what you have.